src/Template/Twig.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Template;
+
+use Twig_Loader_Filesystem as TwigFilesystem;
+use Twig_Environment as TwigEnvironment;
+
+/**
+ * Template implementation bridging league/plates
+ */
+class Twig implements TemplateInterface
+{
+    /**
+     * @var TwigFilesystem
+     */
+    protected $twigLoader;
+
+    /**
+     * @var TwigEnvironment
+     */
+    protected $template;
+
+    public function __construct(TwigEnvironment $template = null)
+    {
+        if (null === $template) {
+            $template = $this->createTemplate();
+        } else {
+            if (!$template->getLoader()) {
+                $this->twigLoader = new TwigFilesystem();
+                $template->setLoader($this->twigLoader);
+            } else {
+                $this->twigLoader = $template->getLoader();
+            }
+        }
```

Comment:
Let's rewrite the above to the following:

```php
if (null === $template) {
    $template = $this->createTemplate();
}

if (! $template->getLoader()) {
    $this->injectDefaultLoader($template);
}

$this->twigLoader = $template->getLoader();
```

`injectDefaultLoader()` would look like this:

```php
private function injectDefaultLoader(TwigEnvironment $template)
{
    $template->setLoader(new TwigFilesystem());
}
```

See notes below for changes to `createTemplate()`.